.github/workflows/deploy-lambda.yml

```diff
@@ -0,0 +1,172 @@
+            OPS_GITHUB_TOKEN=/ci/github/token
+
+      - name: Checkout Code
+        uses: actions/checkout@v3
+        with:
+          repository: 'cmsgov/ab2d-ops'
+          ref: ${{ inputs.ab2dOpsBranch }}
+          token: ${{ env.OPS_GITHUB_TOKEN }}
+
+      - name: Set variables for build version and AB2D environment
+        run: |
+          BUILD_VERSION=$(echo "${{ needs.publish_lambdas.outputs.build_version || inputs.buildVersion }}" | tr -d ' ')
+          echo "BUILD_VERSION=${BUILD_VERSION}"
+          if [ -z "${BUILD_VERSION}" ]; then
+            echo "Unable to determine build version"
+            exit 1
+          fi  
+          
+          if [[ ${{ inputs.environment }} = "dev" ]];         then AB2D_ENV='ab2d-dev'
+          elif [[ ${{ inputs.environment }} = "test" ]];      then AB2D_ENV='ab2d-east-impl'
+          elif [[ ${{ inputs.environment }} = "sbx" ]];       then AB2D_ENV='ab2d-sbx-sandbox'
+          elif [[ ${{ inputs.environment }} = "prod_test" ]]; then AB2D_ENV='ab2d-east-prod-test'
+          elif [[ ${{ inputs.environment }} = "prod" ]];      then AB2D_ENV='ab2d-east-prod'
+          else echo "Unable to determine AB2D environment"; exit 1
+          fi
+          
+          echo "BUILD_VERSION=${BUILD_VERSION}" >> $GITHUB_ENV
+          echo "AB2D_ENV=${AB2D_ENV}" >> $GITHUB_ENV
+
+      - name: Assume role in AB2D account for this environment
+        uses: aws-actions/configure-aws-credentials@v3
+        env:
+          ACCOUNT: ${{ inputs.environment == 'prod_test' && 'prod' || inputs.environment }}
+        with:
+          aws-region: ${{ vars.AWS_REGION }}
+          role-to-assume: arn:aws:iam::${{ secrets[format('{0}_ACCOUNT_ID', env.ACCOUNT)] }}:role/delegatedadmin/developer/ab2d-${{ env.ACCOUNT }}-github-actions
+
+      - name: Download ZIP files
+        env:
+          ARTIFACTORY_PREFIX: 'https://artifactory.cloud.cms.gov/artifactory/ab2d-main/gov/cms/ab2d'
+        run: |
+          mkdir -p terraform/modules/lambda/event-service || true
+          cd terraform/modules/lambda/event-service
+          
+          declare -a lambdas=(
+            'metrics-lambda' 
+            'audit' 
+            'database-management' 
+            'coverage-counts' 
+            'retrieve-hpms-counts'
+          )
+          for name in "${lambdas[@]}"
+            do curl -u "${ARTIFACTORY_USER}:${ARTIFACTORY_PASSWORD}" "${ARTIFACTORY_PREFIX}/${name}/${{ env.BUILD_VERSION }}/${name}-${{ env.BUILD_VERSION }}.zip" --output ${name}.zip
+          done
+          
+          echo "$(ls -lah)"
+          cd -
+
+      # without this step, terraform fails with `/usr/bin/env: ‘node’: No such file or directory`
+      - name: Setup node for terraform
+        uses: actions/setup-node@v2
+        with:
+          node-version: '20'
+
+      - name: Setup terraform
+        uses: hashicorp/setup-terraform@v3
+        with:
+          terraform_version: '1.1.2'
```

Comment:
See my note here for @oluwolenpbc - we should rely on a .terraform-version file instead of hardcoding terraform versions in workflows: https://github.com/CMSgov/ab2d/pull/1446/files#r1987904661